students/semko_krzysztof/lesson_02_flow_control/the_second_maximum.py

```diff
@@ -0,0 +1,28 @@
+"""
+The sequence consists of distinct positive integer numbers
+and ends with the number 0.
+Determine the value of the second largest element in this sequence.
+It is guaranteed that the sequence has at least two elements.
+"""
+
+print("Please input integers in a sequence. to finish, type '0'.")
+
+max_value = 0
+second_max = 0
+loop_number = 1
+while loop_number > 0:
```

Comment:
W tym zadaniu mamy tak długo czytać input dopóki nie wczytamy 0. 
Twoje flow robi coś takiego: jeżeli wczytano 0 => ustaw loop_number na 0 => jeżeli loop_number jest 0, zakończ pętlę (plus jeszcze break, który kończy pętlę tak czy siak).

To sprawia, że używasz loop_number do dwóch rzeczy: zliczania pętli jeśli > 0 i wychodzenia z loopa jeśli <= 0 (przy czym zresetowanie zmiennej do 0 następuje ręcznie). To jest nieczytelny design. W przypadku takiej pętli, która nie wiadomo kiedy się skończy (bo nie wiadomo kiedy wczytamy 0) zalecam jedno z dwóch:

1. `while True` i wyjść przez `break`.
2. dodatkowa zmienna bool np. `processing_done` i wtedy `while processing_done == false`

Wtedy kod sam będzie opowiadał co robi i nie trzeba będzie się domyslać kiedy która zmienna do czego służy :)